### DISCUSSION TITLE
Progress Bar Text above Bar

### DISCUSSION THREAD

**Discussion by rlaphoenix** [Textualize/rich#2830] [Q&A]
I have already found a way by making a Table.grid() and placing the text as one row, and the bar as the other. However, I need each `task` to be in this layout. If I do it this way all I'm actually doing is moving all texts into one group, and the bars into another.

I need:

[task1 text]
[task1 bar]
[task2 text]
[task2 bar]

not the following which can be done with a Table.grid and .add_row():

[task1 text]
[task2 text]
[task1 bar]
[task2 bar]

The reason I need this is because often the progress description value will be wildly too long and will fill up most terminals. Therefore I will need the bar moved to the next line for it to be usable.

**Comment by rlaphoenix** [comment]
```py
table = Table.grid()

# ...

def thread_worker(table):
    table.add_row("Thread...")
    progress = Progress()
    table.add_row(progress)
```

This idea worked for my needs, however, they don't have their columns in the same length and position as each other.

**Comment by willmcgugan** [comment]
I’m afraid I don’t know what you are trying to achieve. A minimal working example would help. Or even a diagram.

**Reply by rlaphoenix** [reply]
![image](https://user-images.githubusercontent.com/17136956/221420881-0d4e1e53-4d04-4768-bea4-753825cdac47.png)
Something along the lines of this is what I was hoping to achieve, via a single Progress() renderable.

While I haven't managed to do so with a Progress() renderable alone, I have been able to by using a Table.grid, and just passing the object to threads and having them use it's add_row to add a progress object when needed. Which as you can see in the image has worked out.

The only downside with this approach is the column width for each Progress column is not going to be kept the same for each progress bar.

**Reply by rlaphoenix** [reply]
Another problem I have is if I have more text+progress bars than I have terminal height, then they get overflowed via a red ellipses. This is not ideal.
